src/transformers/utils/quantization_config.py

```diff
@@ -180,6 +186,111 @@ def update(self, **kwargs):
         return unused_kwargs
 
 
+@dataclass
+class HqqConfig(QuantizationConfigMixin):
+    """
+    Main HqqConfig.
+    Args:
+        nbits (`int`):
+            Number of bits.
+        group_size (`int`):
+            Group-size value.
+        quant_zero (`bool`):
+            Quantize the zero-point.
+        quant_scale (`bool`):
+            Quantize the scaling.
+        offload_meta (`bool`):
+            Offload the meta-data on the CPU.
+        view_as_float (`bool`):
+            View the quantized weight as float (used in distributed training)
+        int (`axis`):
+            Axis along-which grouping is performed.
+        dynamic_config ('dict'):
+            Parameters for dynamic configuration. The key is the name tag of the layer.
+        skip_modules (List[str])::
+            nn,Linear layers to skip.
+        show_progress (bool):
+            Show tqdm quantization for each shard
+        kwargs (`Dict[str, Any]`):
+            Additional parameters from which to initialize the configuration object.
```

Comment:
Could you update the docstring to that the default value is mentioned ? Also could you add the supported values for nbits, group_size and axis ? There is also a typo in the axis parameters.

Reply:
https://github.com/huggingface/transformers/pull/29637/commits/d31837fb7a8f7b7e2a2fabad169fd97d211d226a

Reply:
Thnaks ! Just a few nits left ! Can you add the *optional* flag when the args is not necessarily required. Also can you move the supported values in the description of the args. You can find an example [here](https://github.com/huggingface/transformers/blob/bdbe1662113b49e81c865f0ca9d3922b500c5414/src/transformers/utils/quantization_config.py#L416).

Reply:
Oh I see that you updated it in the latest commit. Let's see if the CI is happy now

Reply:
I added that but ci is still not happy. I am not sure what to do at this point: https://github.com/huggingface/transformers/blob/67c0985d98912e6b7eea3c916978fb847ab52582/src/transformers/utils/quantization_config.py#L186-L212